### PR TITLE
Visitor can see products sorted in different categories

### PR DESCRIPTION
**Pull request by shsteindorsson**
Visitor can see products sorted in different categories
========================================

[PT-link](https://www.pivotaltracker.com/story/show/168829453)

## PR content
In this PR we are adding a feature where products are sorted by categories when listed.

## This PR contains:
- A feature test
- A unit test
- A category model
- Added column to product table
- A new layout for landing page

## What have learned?
- Writing and converting HTML to HAML
- Associations and foreign keys
- Using seeds file to populate a DB

### CODE REVIEW

File: app/models/category.rb [hunk 1]
```diff
@@ -0,0 +1,4 @@
+class Category < ApplicationRecord
+  validates :title, presence: :true
```
Comment (oliverochman) [inline app/models/category.rb]: please use `validates_presence_of`, more readable

File: features/list_categories.feature [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+  In order to choose product more easily
+  I would like to see the products sorted in categories
+
+  Background:
+    Given the following categories exist
+      | title    |
+      | Starters |
+      | Entrees  |
+      | Desserts | 
+    Given the following products exist
```
Comment (oliverochman) [inline features/list_categories.feature:7]: Please change to `And the following products exist`

File: features/list_categories.feature [hunk 3]
```diff
@@ -0,0 +1,25 @@
+Feature: Visitor views list of categorized products
+  As a visitor,
+  In order to choose product more easily
+  I would like to see the products sorted in categories
+
+  Background:
+    Given the following categories exist
```
Comment (oliverochman) [inline features/list_categories.feature]: You are using the `Category.find_or_create_by` in the table step def for creating products. So you do not really need to have this table. A bit cleaner.

**Review by oliverochman — CHANGES_REQUESTED** [review]

**Review by lucamarial — APPROVED** [review]

**Review by AlexSaxena — APPROVED** [review]

**Review by miyeaier — APPROVED** [review]

**Review by ghost — APPROVED** [review]

File: features/list_categories.feature (hunk 2 above)
Comment (shsteindorsson) [reply features/list_categories.feature:7]: When I remove the category table from the `Background`, this one will be `Given` again

File: features/step_definitions/basic_steps.rb [hunk 4]
```diff
@@ -4,6 +4,20 @@
   end
 end
 
+Given("the following categories exist") do |table|
```
Comment (oliverochman) [inline features/step_definitions/basic_steps.rb]: This step is now unnecessary, please remove it

**Review by oliverochman — CHANGES_REQUESTED** [review]

**Review by AlexSaxena — APPROVED** [review]

**Review by oliverochman — APPROVED** [review]

**Pull request merged**